test/jdk/javax/swing/JFileChooser/FileSizeCheck.java

```diff
@@ -21,95 +21,232 @@
+        F1_100_000(1_100_000, "1.1 MB"),
+        F1_100_001(1_100_001, "1.2 MB"),
+
+        F2_800_000(2_800_000, "2.8 MB"),
+
+//        F1_000_000_000(1_000_000_000, "1.0 GB"),
+//        F1_000_000_001(1_000_000_001, "1.1 GB"),
+        ;
+
+        public final String name;
+        public final long size;
+        public final String renderedSize;
+
+        private Path path;
+
+        FileSize(long size, String renderedSize) {
+            this.name = String.format("%03d-%010d.test", ordinal(), size);
+            this.size = size;
+            this.renderedSize = renderedSize;
+        }
+
+        public void create(final Path parent) {
+            path = parent.resolve(name);
+            if (!Files.exists(path)) {
+                try (var f = new RandomAccessFile(path.toFile(), "rw")) {
+                    f.setLength(size);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
                 }
             }
-            fc.setCurrentDirectory(dir.toFile());
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
         }
+
+        public void delete() {
+            try {
+                Files.deleteIfExists(path);
+            } catch (IOException e) {
+                e.printStackTrace();
+                // Don't propagate
+            }
+        }
+    }
+
+    private static JFrame frame;
+    private static JFileChooser fc;
+
+    private static final AtomicReference<String> error = new AtomicReference<>();
+
+    private static void createUI() {
+        // Create temp files
+        Path dir = Paths.get(".");
+        Arrays.stream(FileSize.values())
+              .forEach(f -> f.create(dir));
+
+        fc = new JFileChooser();
+        fc.setControlButtonsAreShown(false);
+        fc.setCurrentDirectory(dir.toFile());
+
+        frame = new JFrame("JFileChooser File Size test");
+        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
         frame.add(fc);
         frame.pack();
+        frame.setLocationRelativeTo(null);
         frame.setVisible(true);
-        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+    }
+
+    private static void disposeUI() {
+        if (frame != null) {
+            frame.dispose();
+        }
     }
 
     public static void main(String[] args) throws InterruptedException,
-            InvocationTargetException {
-        PassFailJFrame passFailJFrame = new PassFailJFrame("JFileChooser Test Instructions",
-                INSTRUCTIONS, 5, 19, 35);
+            InvocationTargetException, AWTException {
         try {
-            SwingUtilities.invokeAndWait(FileSizeCheck::test);
-            passFailJFrame.awaitAndCheck();
+            final Robot robot = new Robot();
+            SwingUtilities.invokeAndWait(FileSizeCheck::createUI);
+
+            robot.waitForIdle();
+            robot.delay(500);
+            SwingUtilities.invokeAndWait(FileSizeCheck::clickDetails);
```

Comment:
Can we click that button programmatically w/o robot? That will make the test a little bit more stable.

Reply:
It is clicked programmatically. https://github.com/openjdk/jdk/blob/1dc4e2b6279f2c0b3180d448c1ce688475c9b025/test/jdk/javax/swing/JFileChooser/FileSizeCheck.java#L221

Robot is used only for `waitForIdle` to ensure events are processed.